sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-nacos/src/main/java/org/apache/shardingsphere/orchestration/center/instance/NacosInstance.java

```diff
@@ -0,0 +1,165 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.orchestration.center.instance;
+
+import com.alibaba.nacos.api.NacosFactory;
+import com.alibaba.nacos.api.config.ConfigService;
+import com.alibaba.nacos.api.config.listener.Listener;
+import com.alibaba.nacos.api.exception.NacosException;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.Executor;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.shardingsphere.orchestration.center.api.ConfigCenter;
+import org.apache.shardingsphere.orchestration.center.configuration.InstanceConfiguration;
+import org.apache.shardingsphere.orchestration.center.listener.DataChangedEvent;
+import org.apache.shardingsphere.orchestration.center.listener.DataChangedEventListener;
+
+/**
+ * The nacos instance for ConfigCenter.
+ *
+ * @author huangjian
+ * @author sunbufu
+ */
+@Slf4j
+public class NacosInstance implements ConfigCenter {
+    
+    private final String defaultGroup = "SHARDING_SPHERE_DEFAULT_GROUP";
+    
+    private ConfigService configService;
+    
+    @Getter
+    @Setter
+    private Properties properties = new Properties();
+    
+    /**
+     * Initialize nacos instance.
+     *
+     * @param config config center configuration
+     */
+    @Override
+    public void init(final InstanceConfiguration config) {
+        try {
+            Properties properties = new Properties();
+            properties.put("serverAddr", config.getServerLists());
+            properties.put("namespace", null == config.getNamespace() ? "" : config.getNamespace());
+            configService = NacosFactory.createConfigService(properties);
+        } catch (final NacosException ex) {
+            log.debug("exception for: {}", ex.toString());
+        }
+    }
+    
+    /**
+     * Get data from nacos instance.
+     *
+     * @param key key of data
+     * @return value of data
+     */
+    @Override
+    public String get(final String key) {
+        return getDirectly(key);
+    }
+    
+    private String getDirectly(final String key) {
+        try {
+            String dataId = key.replace("/", ".");
+            String group = properties.getProperty("group", defaultGroup);
+            long timeoutMs = Long.parseLong(properties.getProperty("timeout", "3000"));
+            return configService.getConfig(dataId, group, timeoutMs);
+        } catch (final NacosException ex) {
+            log.debug("exception for: {}", ex.toString());
+            return null;
+        }
+    }
+    
+    /**
+     * Get node's sub-nodes list.
+     *
+     * @param key key of data
+     * @return sub-nodes name list
+     */
+    @Override
+    public List<String> getChildrenKeys(final String key) {
```

Comment:
If return "null", whether there will be problems.

Reply:
Yes, I think here need @dongzl 's new solution.  😊 But now, I just move the old nacos module in here.